File: solutions/lua/difference-of-squares/1/difference-of-squares.lua
local function square_of_sum(n)
    local sum = 0
    local i;
    for i = 1, n do
        sum = sum + i
    end
    return sum ^ 2
end

local function sum_of_squares(n)
    local squares = 0
    local i
    for i = 1, n do
        squares = squares + i ^ 2
    end
    return squares
end

local function difference_of_squares(n)
    return square_of_sum(n) - sum_of_squares(n)
end

return {
  square_of_sum = square_of_sum,
  sum_of_squares = sum_of_squares,
  difference_of_squares = difference_of_squares
}

File: solutions/ocaml/difference-of-squares/1/difference_of_squares.ml
let square_of_sum x =
  truncate(Float.of_int(x * (x + 1) / 2) ** 2.0)

let sum_of_squares x =
  truncate(Float.of_int(x * (x +1) * (2 * x + 1) / 6))

let difference_of_squares x =
  (square_of_sum x) - (sum_of_squares x)
